### PR TITLE
Add Break screen

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**

### CODE REVIEW

File: app/src/main/java/com/example/mytrainingpal/components/Navigation.kt [hunk 1]
```diff
@@ -34,7 +34,7 @@ sealed class Screen(
         Screen(
             "musclePainMain",
             "Muscle Pain",
-            ShoulderPainIcon(),
```
Comment (MichaelBuessemeyer) [inline app/src/main/java/com/example/mytrainingpal/components/Navigation.kt:37]: @raissacavalcanti Please undo this change. We want the shoulder pain icon instead of a sad smiley :)

File: app/src/main/java/com/example/mytrainingpal/components/Break.kt [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+@Composable
+fun Break(
+    navController: NavController,
+    exerciseList: MutableList<Pair<Exercise, ExerciseDetails>>,
+    currentExerciseIndex: Int
+) {
+    var currentExercise: Int = 4;
+    var totalExercises: Int = 10;
+    var totalBreakTimeInSeconds: Int = 5
+
```
Comment (MichaelBuessemeyer) [inline app/src/main/java/com/example/mytrainingpal/components/Break.kt:41]: Your PR looks like it is almost finished 🎉. At first glance, the onliest thing I could spot is that you use constant values for `currentExercise` and `totalExercises`. 
- `currentExercise` should be `currentExerciseIndex`
- `totalExercises` should be `exerciseList.size`
- Having a constant break time is fine, as the settings screen is currently still missing

Comment (MichaelBuessemeyer) [reply app/src/main/java/com/example/mytrainingpal/components/Break.kt:41]: Nice work @raissacavalcanti

File: app/src/main/java/com/example/mytrainingpal/components/Break.kt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                        setsLeft = exerciseList[currentExerciseIndex].second.sets
+                    )
+                } else {
+                    nextExerciseUp(
+                        gifPath = exerciseList[currentExerciseIndex].first.pathToGif,
+                        exerciseName = exerciseList[currentExerciseIndex].first.name,
+                        suggestedRepsForThatExercise = exerciseList[currentExerciseIndex].second.reps.split(
+                            ","
+                        )[0].toInt(),
+                        setsLeft = exerciseList[currentExerciseIndex].second.sets - (currentExerciseSetCounter + 1)
```
Comment (grittaweisheit) [inline app/src/main/java/com/example/mytrainingpal/components/Break.kt:107]:
```suggestion
                        setsLeft = exerciseList[currentExerciseIndex].second.sets - currentExerciseSetCounter
```

Comment (grittaweisheit) [reply app/src/main/java/com/example/mytrainingpal/components/Break.kt:107]: I would prefer this, since then the leftover sets shown in the break "next exercise" field reflect the actual leftover sets :)

File: app/src/main/java/com/example/mytrainingpal/components/Break.kt [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (grittaweisheit) [inline app/src/main/java/com/example/mytrainingpal/components/Break.kt:197]:
```suggestion
```

File: app/src/main/java/com/example/mytrainingpal/components/Navigation.kt [hunk 5]
```diff
@@ -55,14 +55,18 @@ sealed class Screen(
...
 }
 
 enum class RouteGroups(val route: String) {
     HOME("home"),
     MUSCLE_PAIN("musclePain"),
     TRAINING("training"),
     CALENDAR("calendar"),
-    SETTINGS("settings")
+    SETTINGS("settings"),
+    TOGGLE("toggle")
```
Comment (grittaweisheit) [inline app/src/main/java/com/example/mytrainingpal/components/Navigation.kt:69]: This should actually not be here. The toggle screen belongs to the training group
```suggestion
```

File: app/src/main/java/com/example/mytrainingpal/components/Navigation.kt [hunk 6]
```diff
@@ -55,14 +55,18 @@ sealed class Screen(
 
     object Settings :
         Screen("settingsMain", "Settings", Icons.Default.Settings, RouteGroups.SETTINGS.route)
+    object Toggle :
+        Screen("toggle", "Toggle", Icons.Default.Settings, RouteGroups.TOGGLE.route)
```
Comment (grittaweisheit) [inline app/src/main/java/com/example/mytrainingpal/components/Navigation.kt:59]:
```suggestion
        Screen("toggle", "Toggle", Icons.Default.Settings, RouteGroups.TRAINING.route)
```
see next comment

File: app/src/main/java/com/example/mytrainingpal/components/Navigation.kt [hunk 7]
```diff
@@ -249,6 +253,7 @@ fun AppNavHost(
                     Text("Still Loading View Model")
                 }
             }
+            composable(Screen.Toggle.route) { ToggleScreen(navController, exercises) }
```
Comment (grittaweisheit) [inline app/src/main/java/com/example/mytrainingpal/components/Navigation.kt:256]: coulld you please move this to the trainings group?

Comment (grittaweisheit) [reply app/src/main/java/com/example/mytrainingpal/components/Navigation.kt:256]: Just move it a line up, I think

File: app/src/main/java/com/example/mytrainingpal/components/YoutubeDialog.kt [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+                        Text(text = "Dialog Title")
+                    },
+
+                    confirmButton = {
+                        Button(
+
+                            onClick = {
+                                openDialog.value = false
+                            }) {
+                            Text("This is the Confirm Button")
```
Comment (grittaweisheit) [inline app/src/main/java/com/example/mytrainingpal/components/YoutubeDialog.kt:38]: I would rename the val openDialog, since it is already the name of the parameter
```suggestion
            val showDialog = remember { mutableStateOf(openDialog) }
            if (showDialog.value) {
                AlertDialog(
                    onDismissRequest = {
                        // Dismiss the dialog when the user clicks outside the dialog or on the back
                        // button. If you want to disable that functionality, simply use an empty
                        // onCloseRequest.
                        showDialog.value = false
                    },
                    title = {
                        Text(text = "Dialog Title")
                    },

                    confirmButton = {
                        Button(

                            onClick = {
                                showDialog.value = false
                            }) {
                            Text("This is the Confirm Button")
```

**Review by grittaweisheit — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mytrainingpal/components/YoutubeDialog.kt (hunk 8 above)
Comment (raissacavalcanti) [reply app/src/main/java/com/example/mytrainingpal/components/YoutubeDialog.kt:38]: oh this should be gone 🥲

**Review by grittaweisheit — APPROVED** [review]
